File: apps/Lodgr.Api/Models/InvoiceItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Lodgr.Api.Models;

[Table("invoice_items")]
public class InvoiceItem
{
    [Key]
    [Column("item_id")]
    public long ItemId { get; set; }

    [Column("invoice_id")]
    public long InvoiceId { get; set; }

    [Column("reading_id")]
    public long? ReadingId { get; set; }

    [Column("item_type")]
    public ItemType ItemType { get; set; }

    [StringLength(255)]
    [Column("description")]
    public string? Description { get; set; }

    [Column("quantity", TypeName = "numeric(10,2)")]
    public decimal? Quantity { get; set; }

    [Column("unit_price", TypeName = "numeric(12,2)")]
    public decimal? UnitPrice { get; set; }

    [Column("amount", TypeName = "numeric(12,2)")]
    public decimal Amount { get; set; }
}

public enum ItemType
{
    RENT,
    ELECTRICITY,
    WATER,
    INTERNET,
    GARBAGE,
    PARKING,
    DEPOSIT_CHARGE,
    DEPOSIT_REFUND,
    OTHER,
}

File: apps/api/src/main/java/io/github/hoanghonghuy/lodgr/entity/InvoiceItem.java
package io.github.hoanghonghuy.lodgr.entity;

import io.github.hoanghonghuy.lodgr.enums.InvoiceItemType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

@Entity
@Table(name = "invoice_items")
@Getter
@Setter
public class InvoiceItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "item_id")
    private Long itemId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "invoice_id", nullable = false)
    private Invoice invoice;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reading_id")
    private UtilityReading utilityReading;

    @Enumerated(EnumType.STRING)
    @Column(name = "item_type", nullable = false)
    private InvoiceItemType itemType;

    @Column(name = "description")
    private String description;

    @Column(name = "quantity", precision = 10, scale = 2)
    private BigDecimal quantity;

    @Column(name = "unit_price", precision = 12, scale = 2)
    private BigDecimal unitPrice;

    @Column(name = "amount", nullable = false, precision = 12, scale = 2)
    private BigDecimal amount;
}
